contracts/managers/allocation-pricers/BinaryAllocationPricer.sol

```diff
@@ -0,0 +1,461 @@
+        quoteAssetInstance = _quoteAssetInstance;
+
+        baseAssetOracleInstance = _baseAssetOracleInstance;
+        quoteAssetOracleInstance = _quoteAssetOracleInstance;
+
+        // Query decimals of base and quote assets
+        baseAssetDecimals = _baseAssetInstance.decimals();
+        quoteAssetDecimals = _quoteAssetInstance.decimals();
+
+        // Set Core and setTokenFactory
+        coreInstance = _coreInstance;
+        setTokenFactoryAddress = _setTokenFactoryAddress;
+    }
+
+    /*
+     * Determine the next allocation to rebalance into. If the dollar value of the two collateral sets is more
+     * than 4x different from each other then create a new collateral set. If currently 100% in baseAsset then
+     * a new quote collateral set is created if 0% in baseAsset then a new base collateral set is created.
+     *
+     * @param  _targetBaseAssetAllocation       Target allocation of the base asset
+     * @param  _currentCollateralSet            Instance of current set collateralizing RebalancingSetToken
+     * @return address                          The address of the proposed nextSet
+     * @return uint256                          The USD value of current Set
+     * @return uint256                          The USD value of next Set
+     */
+    function determineNewAllocation(
+        uint256 _targetBaseAssetAllocation,
+        ISetToken _currentCollateralSet
+    )
+        external
+        returns (address, uint256, uint256)
+    {
+        bool toBaseAsset = validateAllocationParams(
+            _targetBaseAssetAllocation,
+            _currentCollateralSet
+        );
+
+        // Create struct that holds relevant information for the currentSet and (potential) nextSet
+        (
+            CollateralSetInfo memory currentSetInfo,
+            CollateralSetInfo memory nextSetInfo
+        ) = getCollateralSetInfo(
+            toBaseAsset,
+            _currentCollateralSet
+        );
+
+        // Check to see if new collateral must be created in order to keep collateral price ratio in line.
+        // If not just return the dollar value of current collateral sets
+        return checkForNewCollateral(
+            currentSetInfo,
+            nextSetInfo,
+            toBaseAsset
+        );
+    }
+
+    /*
+     * Validate passed parameters to make sure target allocation is either 0 or 100 and that the currentSet
+     * was created by core and is made up of the correct component. Finally, return a boolean indicating
+     * whether new allocation should be in baseAsset.
+     *
+     * @param  _targetBaseAssetAllocation       Target allocation of the base asset
+     * @param  _currentCollateralSet            Instance of current set collateralizing RebalancingSetToken
+     * @return boolean                          The address of the proposed nextSet
+     */
+    function validateAllocationParams(
+        uint256 _targetBaseAssetAllocation,
+        ISetToken _currentCollateralSet        
```

Comment:
Note: any collateral Set can be passed in. Even those with multiple components and with any quantity (could be really high or low)

Reply:
You can call this function with a Set that has multiple components with the correct units.

[`baseAsset`, `otherToken`, otherToken`]. This would change every single RB SetToken's collateral to this and try to initiate an auction for this Set

Reply:
Nevermind. It doesn't change it, but there may be unintended consequences of sending a multi-component collateral set in

Reply:
I'll validate that the length is 1